lib/rules/jsx-props-no-spread-multi.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileoverview Prevent JSX prop spreading the same value multiple times
+ * @author Simon Schick
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+const report = require('../util/report');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const messages = {
+  noMultiSpreading: 'Spreading the same value multiple times is forbidden',
+};
+
+/**
+ * Filter for JSON.stringify that omits circular and position structures.
+ *
+ * @param {string} key
+ * @param {*} value
+ * @returns {*}
+ */
+const propertyFilter = (key, value) => (key !== 'parent' && key !== 'range' && key !== 'loc' ? value : undefined);
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Disallow JSX prop spreading the same expression multiple times',
+      category: 'Best Practices',
+      recommended: true,
+      url: docsUrl('jsx-props-no-spread-multi'),
+    },
+    messages,
+  },
+
+  create(context) {
+    return {
+      JSXOpeningElement(node) {
+        const spreads = node.attributes.filter((attr) => attr.type === 'JSXSpreadAttribute');
+        if (spreads.length < 2) {
+          return;
+        }
+        const hashes = new Set();
+        for (const spread of spreads) {
+          // TODO: Deep compare ast function?
+          const hash = JSON.stringify(spread, propertyFilter);
```

Comment:
This is admittedly somewhat of a bandaid.
I used a various of `deepEqual` before to compare all possible matches (which is also O(n^2)).
This performs pretty well and generally has been generating small 'hashes' however I am worried of scenarios where developers create excessively large/syntactically complex spreads.

In these cases we could perform a simple md5/sha hashing though this may be overkill.

Reply:
Optimized it a little by only hashing the argument rather than the full node.

Reply:
Can't the identifier being spread be trivially determinable?

Reply:
The goal here is to also identify complex ast nodes to be identified, (eg. function calls & what not), I'd assume in 99% of the cases this will just be an identifier so the overhead isn't large.

If you think this may be overkill I can remove this functionality.

Reply:
I think that since it can’t ever be complete, it won’t add much value. Also two function calls aren’t guaranteed to return similar objects.

Reply:
One could argue that if you are spreading equivalent function calls (pure or not) you are __probably__ doing something sketchy, though I can see how a plain identifier is the most probable scenario and everything beyond that is too exotic to handle.

I will update this tomorrow (probably).